### PR TITLE
Port features from op-erigon

### PR DESCRIPTION
**Pull request by boyuan-chen**
This PR ports the following features:
1. Hook depositNonce to some RPC methods
2. Add some unit tests
3. Add op-erigon version

### CODE REVIEW

File: eth/stagedsync/stage_mining_exec.go [hunk 1]
```diff
@@ -537,3 +542,110 @@ func NotifyPendingLogs(logPrefix string, notifier ChainEventNotifier, logs types
 	}
 	notifier.OnNewPendingLogs(logs)
 }
+
+// implemented by tweaking UnwindExecutionStage
+func UnwindMiningExecutionStage(u *UnwindState, s *StageState, tx kv.RwTx, ctx context.Context, cfg MiningExecCfg, logger log.Logger) (err error) {
```
Comment (jyellick) [inline eth/stagedsync/stage_mining_exec.go]: Do we understand why this is necessary in the Optimism case while it is not in the Mainnet case?

Comment (boyuan-chen) [reply eth/stagedsync/stage_mining_exec.go]: I am not sure. The added code is used to clean up all stage changes, so it can be the same as the last block.

Comment (jyellick) [reply eth/stagedsync/stage_mining_exec.go]: I'm still struggling to understand why this needs to happen in the Optimism case, but not in the standard Ethereum case.  In both cases, the Ethereum client can propose a block, and in both cases that block can either be confirmed via a forkchoice, or reverted by a forkchoice which does not include that proposed block.  In the case that the forkchoice extends the currently proposed block, no unwinding is necessary.  In the case that the forkchoice is a different block, things are unwound.  Which brings me back to failing to understand why this new path has been added when it is not necessary in the mainnet case.

I'm not saying it's wrong... in fact, I have to believe that there's a good reason for this change, as why would they have implemented it if not.  But I'm also wary of bringing in a large block of code for which we don't understand the necessity and presumably don't have a test to cover.  My initial thought/hope with the stage changes had been that the new stage would fix/enable the pending block gas limit tests, but this looks unrelated to me.

Comment (InoMurko) [reply eth/stagedsync/stage_mining_exec.go]: Is git blame helpful to figure out the reason why this was added?

Comment (boyuan-chen) [reply eth/stagedsync/stage_mining_exec.go]: Based on their comment, it is used for self-reorg. https://github.com/testinprod-io/op-erigon/commit/eab3848ee0e7a644b03a5f6a8445c7af4cc63fbe

Comment (boyuan-chen) [reply eth/stagedsync/stage_mining_exec.go]: Removed for now

**Review by jyellick — APPROVED** [review]

**Review by mmontour1306 — APPROVED** [review]

**Pull request merged**